src/Components/TableItem.js

```diff
@@ -0,0 +1,273 @@
+      return (
+        <option key={category} value={category}>{category}</option>
+      );
+    });
+    const assigneeOptions = this.props.assignees.map(assignee => {
+      return (
+        <option key={assignee} value={assignee}>{assignee}</option>
+      );
+    });
+    const priorityOptions = this.props.priorities.map(priority => {
+      return (
+        <option key={priority} value={priority}>{priority}</option>
+      );
+    });
+    const statusOptions = this.props.status.map(status => {
+      return (
+        <option key={status} value={status}>{status}</option>
+      );
+    });
+
+
+    if (this.state.isEditing) {
+      return (
+        <tr className = {
+          this.handleTrClass(this.props.tableItem.update, this.props.tableItem.status)
+        }>
+          <td>
+            <span>
+              {this.props.tableItem.id}
+            </span>
+          </td>
+          <td>
+            <span>
+              <textarea
+                className="form-control"
+                ref="subject"
+                value={this.state.saveItem.subject}
+                onChange={this.handleChange} >
+              </textarea>
+            </span>
+          </td>
+          <td>
+            <span>
+              <select
+                className="form-control"
+                ref="category"
+                value={this.state.saveItem.category}
+                onChange={this.handleChange} >
+                {categoryOptions}
+              </select>
+            </span>
+          </td>
+          <td>
+            <span>
+              <select
+                className="form-control"
+                ref="assignee"
+                value={this.state.saveItem.assignee}
+                onChange={this.handleChange} >
+                {assigneeOptions}
+              </select>
+            </span>
+          </td>
+          <td>
+            <span>
+              <select
+                className="form-control"
+                ref="priority"
+                value={this.state.saveItem.priority}
+                onChange={this.handleChange} >
+                {priorityOptions}
+              </select>
+            </span>
+          </td>
+          <td>
+            <span>
+              <select
+                className="form-control"
+                ref="status"
+                value={this.state.saveItem.status}
+                onChange={this.handleChange} >
+                {statusOptions}
+              </select>
+            </span>
+          </td>
+          <td>
+            <button 
+              className="btn btn btn-info btn-sm mr-2 save-btn" 
+              onClick={this.saveItem}>
+              Save
+            </button>
+            <button 
+              className="btn btn-outline-dark btn-sm cancel-btn" 
+              onClick={this.cancelEdit}>
+              Cancel
+            </button>
+          </td>
+        </tr>
+      );
+    } else {
```

Comment:
This `if-else` render function is too long, try to extract render functions

https://reactjs.org/docs/conditional-rendering.html